### PR TITLE
 Add /waitfordebugger argument to MGCB.

### PR DESCRIPTION
**Pull request by dellis1972**
The call to `System.Diagnostics.Debugger.Launch` is
not supported on Unix based platforms. This makes
debugging `MGCB.exe` and Content Pipeline extensions
difficult.

So lets add a new argument which will make `MGCB.exe`
wait for a debugger to attach. Also add a extra
update to the `launch.json` to demonstrate how to
attach the `dotnet` debugger to an existing process.

This new system will also work on Windows.

To use this in VSCode you can add the following to the
`launch.json`

```json
{
    // Use IntelliSense to learn about possible attributes.
    // Hover to view descriptions of existing attributes.
    // For more information, visit: https://go.microsoft.com/fwlink/?linkid=830387
    "version": "0.2.0",
    "configurations": [
        {
            "name": "Attach to Process",
            "type": "coreclr",
            "request": "attach",
            "processId": "${input.processid}",
        }
    ],
    "inputs": [
        {
            "id": "processid",
            "type": "promptString",
            "default": "0",
            "description": "Enter Process Id of process to attach to."
        }
    ]
}
```

you can then add the following to your `csproj`.

```
<MonoGameMGCBAdditionalArguments>/waitfordebugger</MonoGameMGCBAdditionalArguments>
```

This will cause the `MGCB.exe` to print the ProcessId
to the console during the build and wait for the debugger
to attach. Once you have the ProcessId you can run the
VSCode `Attach to Process` and enter the ProcessId when prompted.
The debugger should then attach.

If you want to prcess without attaching the debugger just hit any
key in the console or terminal windows where `MGCB.exe` is running.

### CODE REVIEW

File: Tools/MonoGame.Content.Builder/Program.cs [hunk 1]
```diff
@@ -45,6 +47,17 @@ static int Main(string[] args)
                     Console.Error.WriteLine("The debugger is not implemented under Mono and thus is not supported on your platform.");
                 }
             }
+            if (content.WaitForDebuggerToAttach)
+            {
+                var currentProcess = Process.GetCurrentProcess();
+                Console.WriteLine($"Waiting for debugger to attach ({currentProcess.MainModule.FileName} PID {currentProcess.Id}).  Press any key to continue without debugger.");
+                while (!Debugger.IsAttached) {
```
Comment (harry-cpp) [inline Tools/MonoGame.Content.Builder/Program.cs:54]: Braces should be be in new lines.

**Comment by Akarinnnnn** [comment]
`/d` switch also seem to wait for debugger, too.
image

**Comment by dellis1972** [comment]
> `/d` switch also seem to wait for debugger, too. image

That switch will Launch Visual Studio on windows. It does nothing on MacOS/Linux.

File: Tools/MonoGame.Content.Builder/Program.cs (hunk 1 above)
Comment (SimonDarksideJ) [reply Tools/MonoGame.Content.Builder/Program.cs:54]: Agreed, let us use the same braces formatting as the rest of MonoGame, braces on new lines, indented with the function.

Other than that, it is good to go.

**Comment by SimonDarksideJ** [comment]
Is the PR ready for re-review yet @dellis1972 ?

**Comment by dellis1972** [comment]
@SimonDarksideJ it is.

**Review by SimonDarksideJ — APPROVED** [review]
Looks good to me

**Review by SimonDarksideJ — APPROVED** [review]

**Pull request merged**